src/corehost/cli/deps_resolver.cpp

```diff
@@ -539,45 +314,63 @@ void deps_resolver_t::resolve_tpa_list(
         {
             add_tpa_asset(entry.asset_name, candidate, &items, output);
         }
-        // Is this entry present in the secondary package cache?
+        // Is an NI image for this entry present in the secondary package cache?
+        else if (entry.to_hash_matched_path(ni_package_cache_dir, &candidate))
```

Comment:
Shouldnt be matching hash. @anurse @davidfowl  will hash file be present for the NI image?